### ISSUE TITLE
Don't \escape characters inside `code`

### ISSUE THREAD

**Issue by deltaidea** [naokazuterada/MarkdownTOC#64]
Here's a perfectly reasonable header:

``` Markdown
## `function(param, [optional])`
```

Here's what MarkdownTOC generates:

``` Markdown
- [`function\(param, \[optional\]\)`](#functionparam-optional)
```

Here's how it looks:

![image](https://cloud.githubusercontent.com/assets/4950036/14593205/00b644f4-0541-11e6-89c7-4aa811fb419c.png)

Escaping brackets inside `code` is not needed, and those backslashes get rendered.
It should be possible to avoid this problem by using a Markdown parser.

**Comment by jonasbn** [comment]
I have labelled this as an enhancement, one could argue it was a bug, but it does not seem to break use of the plugin, it just renders a less attractive result.

**Comment by math2001** [comment]
Hum... Is this going to be fixed? I might have a look to make a PR if no one wants to work on it (or has the time, although I'm not sure I do either... 😄)

**Comment by naokazuterada** [comment]
@math2001 Hi, thank you for your attention.
I will try to fix this at the next available time.

**Comment by math2001** [comment]
Okay, [it's friday](https://opensourcefriday.com), I'll give it a go. 😄

**Comment by naokazuterada** [comment]
This issue is fixed in #101
